test/acceptance/health-check.js

```diff
@@ -0,0 +1,214 @@
+'use strict';
+
+/**
+ * Module dependencies.
+ */
+
+const assert = require('node:assert');
+const Reader = require('../../lib/reader');
+const Writer = require('../../lib/writer');
+const sinon = require('sinon');
+const uid = require('uid');
+const utils = require('../utils');
+const Connection = require('../../lib/connection');
+
+describe('Acceptance: Health Check', () => {
+  describe('Reader', () => {
+    let topic;
+
+    afterEach(done => {
+      utils.deleteTopic(topic, () => done());
+    });
+
+    beforeEach(done => {
+      topic = uid();
+      utils.createTopic(topic, () => done());
+    });
+
+    describe('using nsqd', () => {
+      it('should send a message with the connections that are expected and active', done => {
+        const reader = new Reader({
+          topic,
+          channel: 'reader',
+          healthCheck: true,
+          nsqd: ['127.0.0.1:4150']
+        });
+
+        reader.on('subscribed', () => {
+          try {
+            const healthCheck = reader.getHealthCheck();
+
+            assert.equal(healthCheck.connections.active, 1);
+            assert.equal(healthCheck.connections.expected, 1);
+
+            reader.end();
+
+            done();
+          } catch (error) {
+            done(error);
```

Comment:
I don't understand the need to invoke `done()` for both success and fail. Feel like it misses the point of actually testing a success of failure case.

Reply:
Invoking `done()` with an argument is basically saying that the test failed. If `done()` is not called, then the test will keep running until it times out.

Reply:
Not very fond of replicating the pattern, but it is what it is.